Thaumaturgy Ritual: Chime of Unseen Spirits

Kolibri-B opened the pull request:
## About The Pull Request

Adds the Thaumaturgy Ritual 'Chime of Unseen Spirits'. This ritual creates an item which can be placed on walls, tables, and floors. When placed properly (completion of the do_after), the chime will ring every 5 seconds when it detects the nearby spirits of Auspex 5 (Psychic Projection) users, and the ghosts of the dead. Does NOT ring for admin ghosts. Currently rings for observers, as they're now considered to be IC wraiths.

Thanks to Rosy for the sprites, and Magisterium for the sound file!

## Why It's Good For The Game

More lore-accurate content is good!

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

<img width="797" height="693" alt="brave_CZ3bAd0DgJ" src="https://github.com/user-attachments/assets/8d5110b2-f163-401b-b26a-e4d733dcb2a4" />

https://github.com/user-attachments/assets/b7c3020e-d723-446e-a02a-c38c640ee002

https://github.com/user-attachments/assets/dff1e519-6a1d-4f6d-b7dc-38119228d054

</details>

## Changelog
:cl:
add: Added new Thaumaturgy ritual: Chime of Unseen Spirits
sound: Added spirit_chime_ring.ogg
image: Added spirit_chime.dmi, with icons "bell" and "chime"
/:cl:

XeonMations commented on modular_tfn/modules/spirit_chime/spirit_chime.dm:
```diff
@@ -0,0 +1,97 @@
+/obj/item/spirit_chime/Initialize()
+	. = ..()
+	spawn(5)
+		loop()
```
Please use an AoE component instead of a constant loop.

XeonMations commented on modular_tfn/modules/spirit_chime/spirit_chime.dm:
```diff
@@ -0,0 +1,97 @@
+		if(istype(M, /mob/dead/observer/avatar)) // Auspex check
+			ring = TRUE
+			break
+		if(istype(M, /mob/dead/observer)) // Ghost check
```
Isavatar() is a check you can use to find auspex users.

Verin-Verdelet commented:
Finally some tremere spirit watching that does not involve sucking on the giovanni teat. 
![APPROVED](https://github.com/user-attachments/assets/ec3f82fe-3337-4592-8ab1-efa44d9b437f)
APPROVED, 1000 TIMES APPROVED.

Kolibri-B commented:
> > > The chimes for observers is terrible and annoying, I get it, they are IC now, but unlike with auspex observers, there's no reason for them to STOP observing you. </3 So it can just kinda ruin it and annoy. Maybe mute the sound? Additionally I've noticed it doesn't pick up on ghosts observing me until I pick it up and back down again, but they may have not been set properly down. I'll test it more and provide info, just wanted to note it here incase anybody else could test it.
> > > THIS ROCKS THOUGH!! I love it so much!
> > 
> > 
> > Regarding it chiming for observers, that can be changed back to how it was initially very simply if that's what people would prefer. I changed it to allow observers to ring due to the ruling regarding observers being IC wraiths now, however if it's too annoying, it can be reverted until wraiths have their own type.
> > As far as not picking up on ghosts observing you until you pick it up and put it back down again, I think the problem here is likely that the chime requires the ghost inside of its field- or the person they're orbiting- to move within the field at least once in order to be detected and tracked. This was, unfortunately, a necessary implementation for performance reasons, as constantly running a loop to check every tile in range is bad and not allowed.
> 
> there's nothing stopping you from having it do a once off check for everyone who should be in the field to be added to the field on creation

That's true, it doesn't solve everything but it _is_ an improvement. I've updated the code to include an initial check on placement as well as blocked it from detecting observers again, since it seems it'd be better if it was just changed once Wraiths are more of a proper thing.
